crates/wasi/src/preview2/ctx.rs

```diff
@@ -250,7 +250,19 @@ impl WasiCtxBuilder {
 
     /// Allow usage of `wasi:sockets/ip-name-lookup`
     pub fn allow_ip_name_lookup(&mut self, enable: bool) -> &mut Self {
-        self.allow_ip_name_lookup = enable;
+        self.allowed_network_uses.ip_name_lookup = enable;
+        self
+    }
+
+    /// Disallow usage of UDP
+    pub fn disallow_udp(&mut self, disable: bool) -> &mut Self {
+        self.allowed_network_uses.udp = !disable;
+        self
+    }
+
+    /// Disallow usage of TCP
+    pub fn disallow_tcp(&mut self, disable: bool) -> &mut Self {
+        self.allowed_network_uses.tcp = !disable;
         self
```

Comment:
Could this be configured similar to the other options here in the positive where it's `allow_{tcp,udp}` instead of `disallow`?

Reply:
I wasn't sure how that wold be feel given that the default is that TCP and UDP are allowed. Given that, most users will do `ctx.allow_tcp(false)` which feels stranger to me than the admittedly inconsistent `ctx.disallow_tcp(true)`. But happy to switch over for consistency sake.

Reply:
It might be more of a personal preference thing on my part then, but I find it easier to read `allow_tcp(false)` than `disallow_tcp(false)` since the latter has two negatives and I have to invert it in my head to realizing that it's allowing TCP. 

I think it's ok to not tie the method name to the default value since whenever it's configured in code it'll be easy to read what's happening and the default would be documented in the documentation as well.